cli/src/generate/prepare_grammar/expand_tokens.rs

```diff
@@ -229,6 +249,14 @@ impl NfaBuilder {
                     if class.negated {
                         chars = chars.negate();
                     }
+                    if self.case_insensitive {
+                        for char in chars.clone().chars() {
+                            let inverted = inverse_char(char);
+                            if char != inverted {
+                                chars = chars.add_char(inverted);
+                            }
+                        }
```

Comment:
I'd wrap this to something like `with_inverse_char(&chars, |inverted| chars = chars.add_char(inverted))` and avoid repetition in other places.